File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.ts
function gcd(a: number, b: number): number {
    while (b !== 0) {
        [a, b] = [b, a % b];
    }
    return a;
}

function findGCD(nums: number[]): number {
    const mn = Math.min(...nums);
    const mx = Math.max(...nums);
    return gcd(mn, mx);
}

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.rs
impl Solution {
    pub fn find_gcd(nums: Vec<i32>) -> i32 {
        let mn = *nums.iter().min().unwrap();
        let mx = *nums.iter().max().unwrap();
        gcd(mn, mx)
    }
}


fn gcd(mut a: i32, mut b: i32) -> i32 {
    while b != 0 {
        let temp = b;
        b = a % b;
        a = temp;
    }
    a
}